--- a/BUILD.gn
+++ b/BUILD.gn
@@ -143,8 +143,12 @@
 # target_defaults and direct_dependent_settings.
 config("common_inherited_config") {
   defines = [ "PROTOBUF_ENABLE_DEBUG_LOGGING_MAY_LEAK_PII=0" ]
-  cflags = []
-  ldflags = []
+  cflags = [ "-fvisibility=default" ]
+  ldflags = [ "-lavutil", "-lavformat", "-lavcodec" ]
+
+  if (is_linux) {
+    ldflags += [ "-Wl,--version-script=" + rebase_path("//libwebrtc.version", root_build_dir) ]
+  }

   if (rtc_objc_prefix != "") {
     defines += [ "RTC_OBJC_TYPE_PREFIX=${rtc_objc_prefix}" ]
